packages/ui/src/components/Text/Text.tsx

```diff
@@ -1,61 +1,68 @@
-import isPropValid from '@emotion/is-prop-valid'
-import styled from '@emotion/styled'
+import clsx from 'clsx'
 import { ReactNode } from 'react'
 import Balancer from 'react-wrap-balancer'
-import { FontSizeProps, getFontSize, theme, UIColors } from '../../theme'
-import { Space } from '../Space'
+import { FontSizeProps } from '../../theme'
+import {
+  textBase,
+  textSprinkles,
+  textStrikethrough,
+  textUppercase,
+  type TextColor,
+  Sprinkles,
+} from './Text.css'
 
-type TextColor = Pick<
-  UIColors,
-  | 'textPrimary'
-  | 'textSecondary'
-  | 'textSecondaryOnGray'
-  | 'textTertiary'
-  | 'textTranslucentPrimary'
-  | 'textTranslucentSecondary'
-  | 'textTranslucentTertiary'
-  | 'textDisabled'
-  | 'textNegative'
-  | 'textGreen'
-  | 'textAmber'
-  | 'textRed'
-  | 'signalBlueText'
-  | 'signalRedText'
-  | 'signalGreenText'
-  | 'signalAmberText'
->
+type TextStyleProps = {
+  align?: Sprinkles['textAlign']
+  balance?: boolean
+  className?: string
+  color?: TextColor
+  size?: FontSizeProps
+  strikethrough?: boolean
+  uppercase?: boolean
+}
```

Comment:
nit:
A suggestion would be to change `color` and `size` type definitions to use `Sprinkles` type. With that we can remove `TextColor` type created inside `Text.css.ts` file and we'd making explicitly that those `props` are being handled by _sprinkles_.
```suggestion
type TextStyleProps = {
  align?: Sprinkles['textAlign']
  balance?: boolean
  className?: string
  color?: Sprinkles['color']
  size?: Sprinkles['fontSize']
  strikethrough?: boolean
  uppercase?: boolean
}
```